File: src/third_iteration/override_methods/Gesture.java
package third_iteration.override_methods;

public class Gesture {

	public static final Rock ROCK = new Rock();
	public static final Paper PAPER = new Paper();
	public static final Scissors SCISSORS = new Scissors();

	private Gesture() {
	}

	static class Rock extends Gesture {
		private Rock() { }
	}
	static class Paper extends Gesture {
		private Paper() { }
	}
	static class Scissors extends Gesture {
		private Scissors() { }
	}
}

File: php/src/third_iteration/override_methods/Gesture.php
<?php

namespace third_iteration\override_methods;

class Gesture
{
    public static function ROCK()
    {
        return new Rock();
    }

    public static function PAPER()
    {
        return new Paper();
    }

    public static function SCISSORS()
    {
        return new Scissors();
    }
}
